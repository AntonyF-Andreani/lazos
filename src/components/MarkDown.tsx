import React from "react";
import ReactMarkdown, { ReactMarkdownOptions } from "react-markdown";

const Markdown = ({ children, ...props }: ReactMarkdownOptions) => (
  <ReactMarkdown {...props}>
    {typeof children === "string"
      ? children.replace(/\n/g, "  \n").replace(/\<\/br>/g, "")
      : children}
  </ReactMarkdown>
);

export default Markdown;
