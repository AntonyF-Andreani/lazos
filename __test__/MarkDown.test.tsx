import React from "react";
import { render } from "@testing-library/react";

import MarkDown from "components/MarkDown";

describe("<Markdown />", () => {
  it("should render without crash", () => {
    const { asFragment } = render(<MarkDown>Test</MarkDown>);

    expect(asFragment()).toMatchSnapshot();
  });
});
