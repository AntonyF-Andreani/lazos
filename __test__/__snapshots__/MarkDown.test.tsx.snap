// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Markdown /> should render without crash 1`] = `
<DocumentFragment>
  <p>
    Test
  </p>
</DocumentFragment>
`;
